librocksdb_sys/crocksdb/c.cc

```diff
@@ -3229,20 +3235,64 @@ void crocksdb_cache_set_capacity(crocksdb_cache_t* cache, size_t capacity) {
   cache->rep->SetCapacity(capacity);
 }
 
-crocksdb_env_t* crocksdb_create_default_env() {
+crocksdb_env_t* crocksdb_default_env_create() {
   crocksdb_env_t* result = new crocksdb_env_t;
   result->rep = Env::Default();
+  result->block_cipher = nullptr;
+  result->encryption_provoider = nullptr;
   result->is_default = true;
   return result;
 }
 
-crocksdb_env_t* crocksdb_create_mem_env() {
+crocksdb_env_t* crocksdb_mem_env_create() {
   crocksdb_env_t* result = new crocksdb_env_t;
   result->rep = rocksdb::NewMemEnv(Env::Default());
+  result->block_cipher = nullptr;
+  result->encryption_provoider = nullptr;
   result->is_default = false;
   return result;
 }
 
+struct CTRBlockCipher : public BlockCipher {
+  CTRBlockCipher(size_t block_size, const std::string& cipertext)
+      : block_size_(block_size), cipertext_(cipertext) {
+    assert(block_size == cipertext.size());
+  }
+
+  virtual size_t BlockSize() { return block_size_; }
+
+  virtual Status Encrypt(char* data) {
+    const char* ciper_ptr = cipertext_.c_str();
+    for (size_t i = 0; i < block_size_; i++) {
+      data[i] = data[i] ^ ciper_ptr[i];
+    }
+
+    return Status::OK();
+  }
+
+  virtual Status Decrypt(char* data) {
```

Comment:
Looks like we don't need to implement this function? How about just abort or assert fail?

Reply:
To keep consistency with the declaration of this API, it's better to implement Decrypt.